tests/EachPromiseTest.php

```diff
@@ -430,4 +441,23 @@ public function testIsWaitableWhenLimited(): void
         $this->assertSame(['a', 'c', 'b', 'd'], $called);
         $this->assertTrue(P\Is::fulfilled($p));
     }
+
+    public function testRewindsExhaustedIterator(): void
```

Comment:
This just tests existing behavior that my first pass at the change broke; figured it was worth adding to document.